src/viem/estimateFeesPerGas.test.ts

```diff
@@ -0,0 +1,27 @@
+import { estimateFeesPerGas } from 'src/viem/estimateFeesPerGas'
+
+describe(estimateFeesPerGas, () => {
+  it('should return the correct fees per gas', async () => {
+    const client = {
+      request: jest.fn(async ({ method, params }) => {
+        expect(params).toBeUndefined()
+        if (method === 'eth_gasPrice') return '0x64' // 100 in hex
+        throw new Error(`Unknown method ${method}`)
+      }),
+    }
+    const fees = await estimateFeesPerGas(client as any)
+    expect(fees).toEqual({ maxFeePerGas: BigInt(100), maxPriorityFeePerGas: undefined })
+  })
+
+  it('when fee currency is specified, it should return the correct fees per gas', async () => {
```

Comment:
Nit about test name: I think it should return the correct fees per gas, regardless of fee currency presence.

Reply:
oh exactly, this was only to highlight the different usage with the extra parameter.
Do you have another proposition?

Reply:
oh no, I just didn't noticed the `expect` in the middle of `client.request` mock!
now it makes sense!